File: shkisme/PGS/Level1/숫자문자열과 영단어.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

int solution(string s) {
    string answer = "";
    map<string, int> myMap;
    myMap["zero"] = 0;
    myMap["one"] = 1;
    myMap["two"] = 2;
    myMap["three"] = 3;
    myMap["four"] = 4;
    myMap["five"] = 5;
    myMap["six"] = 6;
    myMap["seven"] = 7;
    myMap["eight"] = 8;
    myMap["nine"] = 9;
    
    string tmp = "";
    for(char c: s) {
        if('0' <= c && c <= '9'){
            answer += c;
        } 
        else{
          tmp += c;
        } 
        
        if(myMap.find(tmp) != myMap.end()) {
            answer += to_string(myMap[tmp]);
            tmp = "";
        }
    }
    
    return stoi(answer);
}

File: Sjaize/String/숫자 문자열과 영단어.java
// https://school.programmers.co.kr/learn/courses/30/lessons/81301
class Solution {
    private static final String[] words = {
        "zero", "one", "two", "three", "four",
        "five", "six", "seven", "eight", "nine"
    };
    public int solution(String s) {
        for (int i = 0; i < words.length; i++) {
            s = s.replace(words[i], Integer.toString(i));
        }
        return Integer.parseInt(s);
    }
}
